### PR TITLE
Tensor svd

### PR DESCRIPTION
**Pull request by mganahl**
make `tn.svd`  work with `BlockSparseTensor`

### CODE REVIEW

**Review by Thenerdstation — CHANGES_REQUESTED** [review]
Add a test. Otherwise LGTM

**Review by alewis — COMMENTED** [review]
What he said

File: tensornetwork/backends/symmetric/decompositions.py [hunk 1]
```diff
@@ -159,9 +170,26 @@ def svd(
     right_charge_labels = np.empty(0, dtype=np.int16)
     all_ublocks = np.empty(0, dtype=get_real_dtype(tensor.dtype))
     all_vblocks = np.empty(0, dtype=get_real_dtype(tensor.dtype))
+
+  if len(discarded_singvals) > 0:
+    left_discarded_singval_charge_labels = np.concatenate([
+        np.full(discarded_singvals[n].shape[0], fill_value=n, dtype=np.int16)
```
Comment (alewis) [inline tensornetwork/backends/symmetric/decompositions.py:176]: can we break this up?

**Review by alewis — APPROVED** [review]
Looks good to me, except for that one little thing

File: tensornetwork/backends/symmetric/decompositions.py (hunk 1 above)
Comment (mganahl) [reply tensornetwork/backends/symmetric/decompositions.py]: sure thing, is this better?

**Pull request merged**